2023/07/19 code review fixes

adambuttrick opened the pull request.

adambuttrick committed:
Switching to argparse for inputs, adding additional script for splitting all files in a directory

adambuttrick committed:
Update sample_logs.py

adambuttrick committed:
Updating concurrence checks to use params in affiliation request, changing variable names, using threshold in predictor

adambuttrick committed:
Adding affiliation match score logging

adambuttrick committed:
Updating to use argparse and other various improvements

adambuttrick committed:
Update predictor.py

adambuttrick committed:
Changing confidence intervals directory and script to find_optimal_probability_threshold

adambuttrick committed:
Adding to-do to the f-score and removing extraneous space from script name

adambuttrick committed:
Removing errant space from directory name

adambuttrick committed:
Updating variables to remove references to confidence intervals and duplicated assignment

adambuttrick committed:
Update predictor.py

adambuttrick committed:
Update README.MD

dtkaczyk commented on utilities/find_optimal_probability_threshold/find_optimal_probability_threshold.py:
```diff
@@ -0,0 +1,49 @@
+        precision, recall, f1_score = calculate_metrics(
+            true_pos, false_pos, false_neg)
+        results[threshold] = [precision, recall, f1_score]
+    metrics_filename = f"fasttext_threshold_metrics_{datetime.now().strftime('%Y%m%d_%H%M%S')}.csv"
```
It would be good to allow the user to give the output filename as an argument as well. The current name could be the default value.

adambuttrick replied:
Added.

dtkaczyk commented on utilities/find_optimal_probability_threshold/predictor.py:
```diff
@@ -19,12 +19,12 @@ def preprocess_text(self, text):
 		else:
-			return None, None
\ No newline at end of file
+			return None
```
I would suggest returning `None, None` in the `else` branch here. It will be more consistent and should allow you to simplify the code in parse_and_query.py a little bit.

adambuttrick replied:
Updated in https://github.com/ror-community/affiliation-matching-experimental/pull/14/commits/8b60e937443debe4458853b69124f2ccb76af708. Let me know if this is what you had in mine for simplifying.

dtkaczyk replied:
Now I am not sure whether your client code is still correct. After doing:

```
fasttext_prediction = PREDICTOR.predict_ror_id(affiliation, min_fasttext_probability)
```

fasttext_prediction is a tuple with 2 elements. So when you do `if fasttext_prediction:` it will always be true.

What I meant in the original comment is that the change you applied to the values returned by predict_ror_id lets you unpack immediately:

```
predicted_ror_id, prediction_confidence = PREDICTOR.predict_ror_id(affiliation, min_fasttext_probability)
```

and then check only predicted_ror_id in the if.

Thanks to this, it is immediately clear for the reader what exactly is returned by predict_ror_id. Without it, you either unpack the values a few lines later in the code (and by that time the reader will forget where fasttext_prediction variable came from), or you have to use confusing indexes (and to understand what fasttext_prediction[0] is, as a reader I have to consult another file). So I believe unpacking immediately makes the code clearer.

Similar changes should be applied in concurrence_check script as well.

dtkaczyk commented on utilities/concurrence_check/predictor.py:
```diff
@@ -19,12 +19,12 @@ def preprocess_text(self, text):
-			return [label, ratio]
+			return label, ratio
 		else:
 			return None
```
I would suggest returning `None, None` in the `else` branch. It will make the function more consistent and should allow you to simplify the client code. You will be able to unpack the values immediately after calling `predict_ror_id`, like this:

```
ror_id, probability = predict_ror_id(...)
if ror_id is None:
    ...
```

adambuttrick replied:
Updated in https://github.com/ror-community/affiliation-matching-experimental/pull/14/commits/7ddc1b3d3ad5d344a45da3a3681746b3a8497179. Let me know if this is what you had in mine for simplifying.

dtkaczyk commented on utilities/concurrence_check/concurrence_check_top_5.py:
```diff
@@ -16,62 +15,59 @@
-                writer.writerow(reader.fieldnames + ['fasttext_prediction', 'fasttext_confidence',
-                                                     'affiliation_prediction', 'concur', 'affiliation_top_results', 'fasttext_in_affiliation_top_results'])
+                writer.writerow(reader.fieldnames + ['fasttext_prediction', 'prediction_probability', 'affiliation_prediction',
+                                                     'concur', 'affiliation_top_results', 'fasttext_in_affiliation_top_results', 'chosen_score', 'top_scores'])
```
This is a tiny issue, but I would change the order of the fields in the output CSV, just to make sure the fields from ROR affiliation matching are grouped together:

fasttext_prediction, fasttext_probability, ror_aff_prediction, ror_aff_score, ror_aff_top_5, ror_aff_top_5_scores, concur, fasttext_in_ror_aff_top_5

adambuttrick replied:
Added with https://github.com/ror-community/affiliation-matching-experimental/pull/14/commits/475f7ba05c2586add7e20f2feb80a39ae64fe587

adambuttrick committed:
Add metrics file output to find_optimal_probability_threshold.py

https://github.com/ror-community/affiliation-matching-experimental/pull/14#discussion_r1272274133

adambuttrick committed:
Updating predictor to output None tuple and simplifying parse and query

adambuttrick committed:
Updating predictor to output None tuple and concurrence checks

https://github.com/ror-community/affiliation-matching-experimental/pull/14#discussion_r1272323034

adambuttrick committed:
Fixing try-except blocks

adambuttrick committed:
Update concurrence_check_top_5.py to use preferred write order

adambuttrick committed:
Aligning variable names and CSV columns in concurrence_check.py

adambuttrick committed:
Unpacking and output file fixes

adambuttrick committed:
Unpacking fixes
